metals/src/main/scala/scala/meta/internal/decorations/SyntheticsDecorationProvider.scala

```diff
@@ -89,10 +89,7 @@ final class SyntheticsDecorationProvider(
     for {
       focused <- focusedDocument()
       filePath = AbsolutePath(path)
-      sourcePath <-
-        if (!filePath.isScalaFilename)
-          SemanticdbClasspath.toScala(workspace, filePath)
-        else Some(filePath)
+      sourcePath <- SemanticdbClasspath.toScala(workspace, filePath)
```

Comment:
This is now handled inside `toScala` - we don't have `.semanticdb` file for the generated text documents and I think  it's not a good idea to create an artificial one.

Reply:
I've just realized that this logic is a little bit clumsy =)

This method might receive two different types of paths: source file or sematicdb document.
As I see, both call sites are located in `SemanticdbIndexer`: 
- [onChange](https://github.com/scalameta/metals/blob/main/metals/src/main/scala/scala/meta/internal/metals/SemanticdbIndexer.scala#L100) (passes semanticdb)
- [onStandaloneFileChange](https://github.com/scalameta/metals/blob/main/metals/src/main/scala/scala/meta/internal/metals/SemanticdbIndexer.scala#L100) (passes source)

I think it might be better to change `SemanticdbIndexer.onChange` to make it pass the source location and do not change `SemanticdbClasspath.toScala`.
wdyt?

Reply:
Good idea, changed!